helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -139,7 +139,9 @@
   final AtomicReference<Map<String, LiveInstance>> _lastSeenInstances;
   final AtomicReference<Map<String, LiveInstance>> _lastSeenSessions;
 
-  final AtomicReference<Set<String>> _lastSeenCustomizedStateTypes;
+  // map that stores the mapping between instance and the customized state types available on that
+  //instance
+  final AtomicReference<Map<String, Set<String>>> _lastSeenCustomizedStateTypesMap;
```

Comment:
nit, since I am confused again while reviewing the code related to this field, shall we rename it to _lastSeenCustomizedStateTypesRef Or _lastSeenCustomizedStateTypesMapRef?